src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueCreateDialog.kt

```diff
@@ -82,22 +84,46 @@ class IssueCreateDialog(
             openConfigureFieldsDialog(dialogLocator)
         }
 
-        val allFieldsLinkLocator = getVisibleAllFieldsLinkLocator(configureFieldsDialogClass)
-        if (driver.isElementPresent(allFieldsLinkLocator)) {
-            driver.wait(elementToBeClickable(allFieldsLinkLocator)).click()
-            val newAllFieldsLinkLocator = getVisibleAllFieldsLinkLocator(configureFieldsDialogClass)
-            driver.wait(not(elementToBeClickable(newAllFieldsLinkLocator)))
+        if (newConfigureFieldsUi) {
+            val selectLocator = getFieldsSelectLocator(configureFieldsDialogClass);
+
+            if (driver.isElementPresent(selectLocator)) {
+                val s = Select(driver.findElement(selectLocator))
```

Comment:
We can squash this `findElement` call and `isElement` (calling `findElements` underneath) into a single WD call.

Classic style:
```kotlin
val select = driver
    .findElements(selectLocator)
    .singleOrNull()
    ?.let { Select(it) }
if (select != null && select.firstSelectedOption.text != "All Fields") {
    select.selectByVisibleText("All Fields")
    driver.wait(elementToBeClickable(selectLocator));
}
```

Stream style:
```kotlin
driver
    .findElements(selectLocator)
    .singleOrNull()
    ?.let { Select(it) }
    ?.takeIf { it.firstSelectedOption.text != "All Fields" }
    ?.let {
        it.selectByVisibleText("All Fields")
        driver.wait(elementToBeClickable(selectLocator));
    }
```

Reply:
Will do, much clearer.

Reply:
And with smaller WD overhead :slightly_smiling_face: